flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/ResourceDeclaration.java

```diff
@@ -51,7 +51,7 @@ public int getNumNeeded() {
         return numNeeded;
     }
 
-    public Collection<InstanceID> getUnwantedWorkers() {
+    public Set<InstanceID> getUnwantedWorkers() {
```

Comment:
What is the purpose of this change? It is best to add some description in the commit message.

Reply:
unmodifiableCollection does not implement equals method. This make unit test hard to compare whether ResourceDeclaration is the same value.

also updated in commit message